modules/core/src/logic/space-manager.ts

```diff
@@ -611,6 +611,75 @@ export class SpaceManager implements Updateable {
         // reset toUpdateCollisions
         this.toUpdateCollisions.clear();
     }
+
+    // Scan Level Management Methods
+    /**
+     * Set scan level for target (called by job completion)
+     */
+    public setScanLevel(targetId: string, faction: Faction, level: ScanLevel): void {
+        const [target] = this.getObjectPtr(targetId);
+        const factionIndex = Number(faction);
+        if (target && factionIndex >= 0 && factionIndex < Number(Faction.FACTION_COUNT)) {
+            target.scanLevels[factionIndex] = level;
+        }
+    }
+
+    /**
+     * Get scan level for target (used by radar widgets)
+     * Returns ScanLevel.UFO (0) if not set, or at least BASIC for same-faction targets
+     */
+    public getScanLevel(targetId: string, faction: Faction): ScanLevel {
+        const [target] = this.getObjectPtr(targetId);
+        if (!target) {
+            return ScanLevel.UFO;
+        }
+
+        const factionIndex = Number(faction);
+
+        // Objects from same faction have at least BASIC scan level
+        if (target.faction === faction && faction !== Faction.NONE) {
+            return Math.max(target.scanLevels[factionIndex] || ScanLevel.UFO, ScanLevel.BASIC);
```

Comment:
in the same code section, DRY.
extract `target.scanLevels[factionIndex] || ScanLevel.UFO` earlier to a const.